flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/xa/SemanticXidGenerator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.jdbc.xa;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.functions.RuntimeContext;
+
+import javax.transaction.xa.Xid;
+
+import java.security.SecureRandom;
+
+/**
+ * Generates {@link Xid} from:
+ * <ol>
+ *     <li>checkpoint id</li>
+ *     <li>subtask index</li>
+ *     <li>4 random bytes to provide uniqueness across other jobs and apps (generated at startup using {@link SecureRandom})</li>
+ * </ol>
+ * Each {@link SemanticXidGenerator} instance MUST be used for only one Sink (otherwise Xids could collide).
+ */
+@Internal
+class SemanticXidGenerator implements XidGenerator {
```

Comment:
What does "Semantic" mean in this class?

Reply:
It generates XIDs based on subtask index and checkpoint, as opposed to fully random or sequential transaction IDs for example in Kafka.
Revisiting this class, I think that `RuntimeContextXidGenerator` would probably be a better name.
WDYT?